core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -164,6 +164,7 @@ public void writePayload(InputStream message) {
   }
 
   private int writeUncompressed(InputStream message, int messageLength) throws IOException {
+    verifyNotClosed();
```

Comment:
No. This will cause exceptions for normal usage of the API. Given the interaction with cancel, these should just return early. And we already have essentially the same thing in AbstractStream. (Neither approach is thread-safe though.)

Reply:
I changed `verifyNotClosed()` to check `isClosed()` and return early if the framer is closed.